crypto/x509/x509_lu.c

```diff
@@ -598,14 +598,17 @@ int X509_STORE_CTX_get1_issuer(X509 **issuer, X509_STORE_CTX *ctx, X509 *x) {
   X509_OBJECT obj, *pobj;
   int idx, ret;
   size_t i;
+  *issuer = NULL;
   xn = X509_get_issuer_name(x);
   if (!X509_STORE_CTX_get_by_subject(ctx, X509_LU_X509, xn, &obj)) {
     return 0;
   }
   // If certificate matches all OK
   if (ctx->check_issued(ctx, x, obj.data.x509)) {
-    *issuer = obj.data.x509;
-    return 1;
+    if (x509_check_cert_time(ctx, obj.data.x509, /*suppress_error*/1)) {
+      *issuer = obj.data.x509;
```

Comment:
Not obvious at quick glance, is there a reason why this reference doesn't need to be bumped?

Reply:
A ref count gets bumped in [`X509_STORE_CTX_get_by_subject`](https://github.com/aws/aws-lc/blob/044d529f6c84c30849a2beff60b1660932f31c93/crypto/x509/x509_lu.c#L311).  I think we would only need to "free" if we return w/o setting the `*issuer` to point to this value, which it does below.

Reply:
@justsmth had the same question. 

The code is pretty confusing. 
Step 1: We define `obj` at the top on L598 and `X509_STORE_CTX_get_by_subject` does a reference bump on the retrieved `obj` if the function succeeds. 
Step 2: If successful, the data within `obj` is then assigned to `issuer` and returned. Technically the reference bump has been done `X509_STORE_CTX_get_by_subject`, so we don't need to do an additional bump here.
Step 3: If failed, we free `obj` which contains the bumped reference on L613.

Reply:
Ah I should have expanded the resolved comments. Thanks!